impl/src/main/java/org/jboss/weld/Container.java

```diff
@@ -208,4 +216,21 @@ public void setState(ContainerState state) {
         this.state = state;
     }
 
+    /**
+     * Returns {@link Environment} that this container was booted in, or null if not set or invoked too
+     * early in the bootstrapping process.
+     *
+     * Note that should there be multiple containers active, this will return the {@link Environment} that the first container
+     * was bootstrapped in, unless it was null.
+     *
+     * @return {@link Environment} or null
+     */
+    public static Environment getEnvironment() {
+        return environmentReference.get();
+    }
+
+    private static void setEnvironment(Environment env) {
```

Comment:
Do we actually need this method? I think we could use `Container.environmentReference.compareAndSet(null, env)` directly in constructor...

Reply:
That's what I originally had, but Spotbugs complained that I'm setting value directly to static field.
That was before using `AtomicReference` though, should be ok now, let's see...